django/middleware/http.py

```diff
@@ -15,6 +18,9 @@ def process_response(self, request, response):
         if not response.streaming and not response.has_header('Content-Length'):
             response['Content-Length'] = str(len(response.content))
 
+        if settings.USE_ETAGS and not response.has_header('ETag'):
```

Comment:
Seems like we may need a deprecation warning if `ConditionalGetMiddleware` is used but `settings.USE_ETAGS=False` since once the deprecation period ends, the behavior will change. It might be friendlier to use the system check framework (django/core/checks/compatibility) for the warnings. That way we can give the warnings up front and not have to wait until a request (and trigger a warning for most or every one).